### PR TITLE
MGMT-8438: UI for disk encryption - day 1

### PR DESCRIPTION
**Pull request by DanielleChason**
I added the ui for the disk encryption from that epic: https://issues.redhat.com/browse/MGMT-8438

This feature lets the user add Disk Encryption for masters and/or workers. 
The user can choose between two encryption modes - tpmv2 and tang. 
If he chooses the tang mode then he must enter at least one pair of Server URL+ Thumbprint. 
Currently it's not possible to choose a different encryption mode for workers and masters. This option can be added after the BE will provide a support for that. 
After creating the cluster the user can't change the encryption (same as SNO). 

https://user-images.githubusercontent.com/77008341/149136733-443c3abf-4f87-4ccf-b86c-ee0383003c4e.mp4

image

### CODE REVIEW

**Comment by nirfarkas** [comment]
UI:
- The "?" dropped down, they should be aligned with the label.
- In the screenshot the selected option is Tang and you're showing an alert for TPM.
- I'm missing the popovers for both TPM and Tang

Text:
Let's change the text in the inline alert to:
“To use this encryption, enable TPM v2 encryption in the BIOS of each host selected”

**Comment by DanielleChason** [comment]
> UI:
> 
> * The "?" dropped down, they should be aligned with the label.
> * In the screenshot the selected option is Tang and you're showing an alert for TPM.
> * I'm missing the popovers for both TPM and Tang
> 
> Text: Let's change the text in the inline alert to: “To use this encryption, enable TPM v2 encryption in the BIOS of each host selected”

About the Tang - I was asking about that, but I understand now what I need to do with that (it didn't apear in the original task). I'm doing it now.  

What popovers do you mean? In the prototype here I see only the "?" which I implemented:
image

**Comment by jkilzi** [comment]
TANG method requires the user to enter 2 values for each TANG server:
1. The TANG server `url`
2. The TANG server `thumbprint` 

**The current design doesn't include the second field.**

/cc @nirfarkas 
/cc @ybettan

File: src/common/components/clusterConfiguration/DiskEncryptionFields.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        </StackItem>
+      )}
+      <StackItem>
+        <SwitchField
+          name="enableDiskEncryptionOnWorkers"
+          isDisabled={isDisabled}
+          label="Enable encryption of installation disks workers"
+        />
+      </StackItem>
+      {(enableDiskEncryptionOnWorkers || enableOnBoth) && (
```
Comment (jkilzi) [inline src/common/components/clusterConfiguration/DiskEncryptionFields.tsx:40]:
```suggestion
      {(enableDiskEncryptionOnWorkers) && (
```
This condition is equivalent to just `enableDiskEncryptionOnWorkers` (Absorbtion law: `A or (A and B) === A` )

File: src/common/components/clusterConfiguration/DiskEncryptionFields/DiskEncryptionMode.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        }
+      />
+    </>
+  );
+};
+
+const DiskEncryptionModeTang: React.FC = () => {
+  return (
+    <>
+      Tang{' '}
```
Comment (jkilzi) [inline src/common/components/clusterConfiguration/DiskEncryptionFields/DiskEncryptionMode.tsx:44]:
```suggestion
      TANG{' '}
```

File: src/common/components/clusterConfiguration/DiskEncryptionFields/DiskEncryptionMode.tsx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        }
+      />
+    </>
+  );
+};
+
+const DiskEncryptionModeTang: React.FC = () => {
+  return (
+    <>
+      TANG{' '}
```
Comment (jkilzi) [inline src/common/components/clusterConfiguration/DiskEncryptionFields/DiskEncryptionMode.tsx:44]:
```suggestion
      Tang{' '}
```

**Pull request merged**